sgkit_plink/pysnptools.py

```diff
@@ -137,18 +136,20 @@ def read_plink(
     lock: bool = False,
     persist: bool = True,
 ) -> Dataset:
-    """Read PLINK dataset
+    """Read PLINK dataset.
 
     Loads a single PLINK dataset as dask arrays within a Dataset
     from bed, bim, and fam files.
 
     Parameters
     ----------
-    path : PathType
-        Path to PLINK file set.
-        This should not include a suffix, i.e. if the files are
-        at `data.{bed,fam,bim}` then only 'data' should be
-        provided (suffixes are added internally).
+    path : Union[PathType, Tuple[PathType, PathType, PathType]]
+        Path to PLINK file set.  This can either be one path or 3.
+        If one path, then that path should not include a suffix such
+        that if the files are at `data.{bed,bim,fam}` then only 'data'
+        should be provided (suffixes are added internally). If 3 paths,
```

Comment:
Depending on an order in a tuple seems rather error prone, wouldn't it be more idiomatic to have 4 flags:

```
bed_path: Optional[PathType],
bim_path: Optional[PathType],
fam_path: Optional[PathType],
path_prefix: Optional[PathType],  # {bed,bim,fam} are added

# check either all([bed_path, bim_path, fam_path]) or path_prefix
```

and with that order you can still do:

```python
read_plink(*paths, sep=blah)
```

?

Reply:
It looks like this change is independent of the pysnptools upgrade? If so, then it's the kind of change that could go in a separate PR.

Reply:
I agree with @tomwhite - this is a significant UI decision which is independent of the pysnptools upgrade.

Reply:
I see it as trivial this early but happy to pull it out: https://github.com/pystatgen/sgkit-plink/pull/15